pennylane/templates/subroutines/gqsp.py

```diff
@@ -115,9 +114,10 @@ def _flatten(self):
     def _unflatten(cls, data, metadata):
         return cls(unitary=metadata[0], angles=data[0], control=metadata[1])
 
+    # pylint: disable=arguments-differ
     @classmethod
-    def _primitive_bind_call(cls, *args, **kwargs):
-        return cls._primitive.bind(*args, **kwargs)
+    def _primitive_bind_call(cls, unitary, angles, control, id=None):
+        return super()._primitive_bind_call(angles, unitary=unitary, wires=control, id=id)
```

Comment:
The `unitary` is an operator, and therefore needs to be treated as potentially trainable.